dm/master/config.go

```diff
@@ -228,3 +288,59 @@ func (c *Config) Reload() error {
 
 	return c.adjust()
 }
+
+// genEmbedEtcdConfig generates the configuration needed by embed etcd.
+func (c *Config) genEmbedEtcdConfig() (*embed.Config, error) {
+	cfg := embed.NewConfig()
+	cfg.Name = c.Name
+	cfg.Dir = c.DataDir
+
+	// reuse the previous master-addr as the client listening URL.
+	cURL, err := parseURLs(c.MasterAddr)
+	if err != nil {
+		return nil, terror.ErrMasterGenEmbedEtcdConfigFail.Delegate(err, "--listen-client-urls/--advertise-client-urls")
+	}
+	cfg.LCUrls = cURL
+	cfg.ACUrls = cURL
+
+	cfg.LPUrls, err = parseURLs(c.PeerUrls)
+	if err != nil {
+		return nil, terror.ErrMasterGenEmbedEtcdConfigFail.Delegate(err, "--listen-peer-urls")
+	}
+
+	cfg.APUrls, err = parseURLs(c.AdvertisePeerUrls)
+	if err != nil {
+		return nil, terror.ErrMasterGenEmbedEtcdConfigFail.Delegate(err, "--initial-advertise-peer-urls")
+	}
+
+	cfg.InitialCluster = c.InitialCluster
+
+	return cfg, nil
+}
+
+// parseURLs parse a string into multiple urls.
+// if the URL in the string without protocol scheme, use `http` as the default.
+// if no IP exists in the address, `0.0.0.0` is used.
+func parseURLs(s string) ([]url.URL, error) {
+	if s == "" {
+		return nil, nil
+	}
+
+	items := strings.Split(s, ",")
+	urls := make([]url.URL, 0, len(items))
+	for _, item := range items {
+		u, err := url.Parse(item)
+		if err != nil && (strings.Contains(err.Error(), "missing protocol scheme") ||
+			strings.Contains(err.Error(), "first path segment in URL cannot contain colon")) {
```

Comment:
how about adding a comment to show two examples, in which two cases will generate these error?

Reply:
add some comments in https://github.com/pingcap/dm/pull/344/commits/31436e8155c1fac9c59e8a2bc5d78c8399d651a6.;